### DISCUSSION TITLE
Error: Invalid access of closed

### DISCUSSION THREAD

**Discussion by kvalood** [uNetworking/uWebSockets.js#793] [General]
I don't quite understand how to fix this problem. Tell me please. 

```
ws.send(JSON.stringify({uuid: ws.uuid}))
                        ^
 Error: Invalid access of closed uWS.WebSocket/SSLWebSocket.
     at open (/index.js:906:24)
     at runMicrotasks (<anonymous>)
     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by e3dio** [comment]
The ws was closed. Looks like you have async open event, then you should add ws.closed=true in your close event and check it after doing any async

**Reply by kvalood** [reply]
> Вс закрыли. Похоже, у вас есть асинхронное событие открытия, тогда вы должны добавить ws.closed=true в событие закрытия и проверить его после выполнения любого асинхронного

does the attempt to send a message happen after the socket is closed?

**Reply by e3dio** [reply]
Yes you tried to send a message after it closed. Set `ws.closed = true` in the `close` event, and check `ws.closed` after doing async before sending message
